packages/components/src/DatePicker/DatePicker.test.tsx

```diff
@@ -111,6 +111,28 @@ describe("Ensure ReactDatePicker CSS class names exists", () => {
   });
 });
 
+describe("Smart autofocus", () => {
+  it("should focus on the selected date", async () => {
+    const { getByTestId } = render(
+      <DatePicker selected={new Date()} onChange={jest.fn} />,
+    );
+    await popperUpdate(() => fireEvent.click(getByTestId("calendar")));
+
+    expect(document.activeElement).toHaveClass(
+      "react-datepicker__day--selected",
+    );
```

Comment:
Couple questions here:

Should there be
```
    expect(document.activeElement).toHaveFocus();
 ```
 
 Also, does it make sense to check if the calendar icon/activator is focused again once a date is selected?

Reply:
> Should there be the following check in this test:

I'll let you check what `document.activeElement` is and then ask yourself that same question 😉 

> Also, does it make sense to check if the calendar icon/activator is focused again once a date is selected?

Already tested by the hook itself

Reply:
```
I'll let you check what document.activeElement is and then ask yourself that same question 😉
```
The reason why I asked is because even if the date is not "focused" and just "selected" it will still have the `react-datepicker__day--selected` class.